### DISCUSSION TITLE
Working, non-working, Incremental Static Regeneration on Heroku

### DISCUSSION THREAD

**Discussion by u840903** [vercel/next.js#17080] [Help] [answered]
Hey!

I've got a project with incremental static regeneration. If I locally boot-up next using `next start` and hit the URL `http://localhost:3000/products/123` the files `/.next/server/pages/products/123.html` and `/.next/server/pages/products/123.json` are created and everything feels nice and speedy.

If I deploy the very same application on Heroku and hit the URL `http://my-app.heroku.com/products/123`, login in to my environment using the heroku cli, `heroku run bash`, `cd /.next/server/pages/products/` there are **no** files created, but it still feels nice and speedy.

Why is that?

1) Are the files created, but I can't see them?
2) Does Next use some kind of in-memory store if it does not have file-write-access?
3) Does ISR not work out-of-the-box on Heroku, but my server is fast enough anyway.
4) Should I use Vercel to get the most out of Next?

Thanks in advance,
Jan

**Comment by birdcar** [comment, marked as answer]
Hi @u840903 👋 

The reason you're not seeing the files when you log into your environment using the Heroku CLI (my guess is using `heroku run bash` or something similar) is because all processes in a Heroku application run in their own Dyno, and each Dyno is a sandboxed container.

As a result, the files generated by your `web` process's Dyno are likely created just as you expect them to be, but you can't see them because when you run `heroku run bash` you actually spin up a totally separate Dyno that hasn't generated those files.

Hopefully that clarifies things!

**Reply by u840903** [reply]
Oh! I see. Thank you!
